lib/Sema/CSDiag.h

```diff
@@ -26,7 +26,14 @@ namespace swift {
   /// UnresolvedType.
   Type replaceTypeParametersWithUnresolved(Type ty);
   Type replaceTypeVariablesWithUnresolved(Type ty);
-  
+
+  /// Diagnose lvalue expr error.
+  void diagnoseSubElementFailure(Expr *destExpr,
+                                        SourceLoc loc,
```

Comment:
My plan is that once all rvalue-as-lvalue errors are handled through this FailureDiagnostic then this function can become private to the `RValueTreatedAsLValueFailure` class. (And be renamed, because the name is pretty poor at the moment... it is determining which sub element of the `destExpr` is what makes it get-only and putting together a diagnostic string for it.) 

So this is only here temporarily. (And yeah, will fix indentation) :)